### PR TITLE
Changelog 2023.10

### PR DESCRIPTION
**Pull request by lang-m**

### CODE REVIEW

File: source/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,86 @@
...
+  - Renamed and modified: ``plane`` has been replaced with a new ``sel`` method. Refer to the documentation for more details.
+  - Removed: ``attributes``
+  - Removed: ``dS`` (used for integrals) because a new syntax for integrals is used
+  - Renamed: ``midpoints`` to ``points``
+  - Modified: new method ``coordinate_field`` that was previous part of the ``Field`` class.
+  - New: ``scale`` and ``translate`` similar to the ``Region`` class.
+
+``Field``
+  - Modified: getting the value at a point (``field(...)``) now returns a ``numpy.ndarray``.
+  - Modified: only field values are returned when iterating over a field.
```
Comment (kzqureshi) [inline source/changelog.rst:49]: add more details please

File: source/changelog.rst [hunk 2]
```diff
@@ -2,6 +2,86 @@
...
+  - Modified: ``angle`` now takes a vector as reference and does not anymore require sliced fields.
+  - Renamed: ``dim`` -> ``nvdim`` to avoid ambiguities, the new name is also required as an argument when creating a field.
+  - Renamed: ``components`` -> ``vdims``
+  - Renamed: ``units`` -> ``unit``
+  - Removed: ``value``, to update field values use the new method ``update_field_values`` instead, to read data use the ``array`` property.
+  - Removed: ``zero`` class method; omit the ``value`` argument to create a field filled with zeros.
+  - New: method ``resample`` to compute field values on a new mesh; the nearest points will be used, interpolation is not supported.
+  - New: property ``valid`` to define parts that contain material; this can be set to the field norm on init using the optional ``valid='norm'`` argument.
+  - Modified: new hdf5 format to store all field information (e.g. subregions were missing before).
+
```
Comment (kzqureshi) [inline source/changelog.rst:66]: integration is not mentioned

File: source/changelog.rst [hunk 3]
```diff
@@ -2,6 +2,116 @@
...
+``discretisedfield`` has been refactored to support scalar and (n-dimensional)
+vector fields on n-dimensional meshes. As part of this refactoring several
+inconsistencies and ambiguities have been removed. These changes break existing
+code! The following summary lists all user-facing changes (removed, renamed, added functionality).
+
+The shape of the domain (i.e. the shape of the region/mesh) is now referred to
+as ``dim``, the shape of the field values as ``vdim`` (vector/value dimension).
+
+``Region``
+  - Modified: scaling a region is now done with the method ``scale``, the option to scale a region by multiply it with a number has been removed.
```
Comment (lang-m) [inline source/changelog.rst:20]:
```suggestion
  - Modified: scaling a region is now done with the method ``scale``, the option to scale a region by multiply it with a number has been removed. Scaling can optionally be done in-place.
```

File: source/changelog.rst [hunk 4]
```diff
@@ -2,6 +2,116 @@
...
+  - New: regions can be translated in space with ``translate``.
```
Comment (lang-m) [inline source/changelog.rst:21]:
```suggestion
  - New: regions can be translated in space with ``translate``.  Translating can optionally be done in-place.
```

File: source/changelog.rst [hunk 5]
```diff
@@ -2,6 +2,116 @@
...
+  - Modified: a new method ``allclose`` checks if two regions are the same within numerical accuracy, the operator now ``==`` checks for exact equality.
+  - Renamed: the operator ``|`` has been renamed to ``facing_surface``
```
Comment (lang-m) [inline source/changelog.rst:23]:
```suggestion
  - Modified: the operator ``|`` has been replaced with the new method ``facing_surface``
```

File: source/changelog.rst [hunk 6]
```diff
@@ -2,6 +2,116 @@
...
+  - Removed: properties ``p1`` and ``p2``, use ``pmin`` and ``pmax``
+  - Changed: the return type of points (e.g. ``pmin``, ``pmax``, ``centre``) is now a ``numpy.ndarray``
+  - Changed: ``unit`` has been renamed to ``units`` and is a tuple to supports different units along different directions (only used as labels, not for calculations).
```
Comment (lang-m) [inline source/changelog.rst:26]:
```suggestion
  - Changed: ``unit`` has been renamed to ``units`` and is a tuple to supports different units along different directions (only used as labels on plots, not for calculations).
```

File: source/changelog.rst [hunk 7]
```diff
@@ -2,6 +2,116 @@
...
+  - New: method ``rotate90`` to perform 90 degree rotations of the region.
+
+``Mesh``
+  - Modified: a new method ``allclose`` checks if two regions are the same
+    within numerical accuracy, the operator now ``==`` checks for exact
+    equality.
+  - Modified: inconsistent behaviour of ``mesh[subregion]``; this operator will
+    now always use closed intervals, i.e. the returned submesh is now always
+    inclusive of endpoints.
+  - Renamed: the operator ``|`` is now a method called ``is_aligned``.
```
Comment (lang-m) [inline source/changelog.rst:38]:
```suggestion
  - Modified: the operator ``|`` has been replaced with a new method called ``is_aligned``.
```

File: source/changelog.rst [hunk 8]
```diff
@@ -2,6 +2,116 @@
...
+  - Removed: ``axispoints`` (was deprecated), use ``points`` instead.
+  - Removed: ``neighbours``
+  - Renamed and modified: ``plane`` has been replaced with a new ``sel`` method. Refer to the documentation for more details.
```
Comment (lang-m) [inline source/changelog.rst:41]: add link to documentation

File: source/changelog.rst [hunk 9]
```diff
@@ -2,6 +2,116 @@
...
+  - Removed: ``attributes``
+  - Removed: ``dS`` (used for integrals) because a new syntax for integrals is used
```
Comment (lang-m) [inline source/changelog.rst:43]: link to documentation?

File: source/changelog.rst (hunk 1 above)
Comment (lang-m) [reply source/changelog.rst:51]: add link to documentation

File: source/changelog.rst [hunk 10]
```diff
@@ -2,6 +2,116 @@
...
+  - Renamed: ``midpoints`` to ``points``
+  - Modified: new method ``coordinate_field`` that was previous part of the ``Field`` class.
+  - New: ``scale`` and ``translate`` similar to the ``Region`` class.
+  - New: method ``rotate90`` to perform 90 degree rotations of the mesh.
+
+``Field``
+  - Modified: getting the value at a point (``field(...)``) now returns a ``numpy.ndarray``.
+  - Modified: only field values are returned when iterating over a field.
+  - Modified: ``coordinate_field`` is now part of the ``Mesh`` class.
```
Comment (lang-m) [inline source/changelog.rst:52]:
```suggestion
  - Removed: ``coordinate_field`` is now part of the ``Mesh`` class.
```

File: source/changelog.rst [hunk 11]
```diff
@@ -2,6 +2,116 @@
...
+  - Renamed and modified: ``derivative`` -> ``diff``; now supports holes inside the field, refer to the documentation for details.
```
Comment (lang-m) [inline source/changelog.rst:53]: add link to documentation

File: source/changelog.rst [hunk 12]
```diff
@@ -2,6 +2,116 @@
...
+  - Modified: new integration syntax, refer to the documentation for details.
```
Comment (lang-m) [inline source/changelog.rst:54]: add link to documentation

File: source/changelog.rst [hunk 13]
```diff
@@ -2,6 +2,116 @@
...
+  - Renamed: ``fromfile`` -> ``from_file``
+  - Renamed and modified: ``plane`` has been replaced with a new ``sel`` method. Refer to the documentation for more details.
```
Comment (lang-m) [inline source/changelog.rst:56]: add link to documentation

File: source/changelog.rst [hunk 14]
```diff
@@ -2,6 +2,116 @@
...
+  - Removed: ``project`` has been removed, use ``mean`` with the ``axis`` argument instead.
+  - Renamed: ``write`` -> ``to_file``
+  - Modified: ``angle`` now takes a vector as reference and does not anymore require sliced fields.
+  - Renamed: ``dim`` -> ``nvdim`` to avoid ambiguities, the new name is also required as an argument when creating a field.
+  - Renamed: ``components`` -> ``vdims``
+  - Renamed: ``units`` -> ``unit``
+  - Removed: ``value``, to update field values use the new method ``update_field_values`` instead, to read data use the ``array`` property.
+  - Removed: ``zero`` class method; omit the ``value`` argument to create a field filled with zeros.
```
Comment (lang-m) [inline source/changelog.rst:64]:
```suggestion
  - Removed: ``zero`` class method; omit the ``value`` argument to create a field filled with zeros (was already supported before).
```

File: source/changelog.rst [hunk 15]
```diff
@@ -2,6 +2,116 @@
...
+  - New: method ``resample`` to compute field values on a new mesh; the nearest points will be used, interpolation is not supported.
+  - New: property ``valid`` to define parts that contain material; this can be set to the field norm on init using the optional ``valid='norm'`` argument.
```
Comment (lang-m) [inline source/changelog.rst:66]: link to documentation

File: source/changelog.rst [hunk 16]
```diff
@@ -2,6 +2,116 @@
...
+  - Renamed and modified: ``plane`` has been replaced with a new ``sel`` method. Refer to the documentation for more details.
+  - Removed: ``project`` has been removed, use ``mean`` with the ``axis`` argument instead.
+  - Renamed: ``write`` -> ``to_file``
+  - Modified: ``angle`` now takes a vector as reference and does not anymore require sliced fields.
+  - Renamed: ``dim`` -> ``nvdim`` to avoid ambiguities, the new name is also required as an argument when creating a field.
+  - Renamed: ``components`` -> ``vdims``
+  - Renamed: ``units`` -> ``unit``
+  - Removed: ``value``, to update field values use the new method ``update_field_values`` instead, to read data use the ``array`` property.
+  - Removed: ``zero`` class method; omit the ``value`` argument to create a field filled with zeros.
+  - New: method ``resample`` to compute field values on a new mesh; the nearest points will be used, interpolation is not supported.
```
Comment (lang-m) [inline source/changelog.rst:65]: link to documentation?

File: source/changelog.rst (hunk 2 above)
Comment (lang-m) [reply source/changelog.rst:79]: Done

File: source/changelog.rst [hunk 17]
```diff
@@ -2,6 +2,116 @@
...
+  - New: property ``valid`` to define parts that contain material; this can be set to the field norm on init using the optional ``valid='norm'`` argument.
+  - Modified: new hdf5 format to store all field information (e.g. subregions were missing before).
```
Comment (lang-m) [inline source/changelog.rst:67]: link to format specification

**Review by lang-m — COMMENTED** [review]
@samjrholt Can you please look at my suggested changes and apply those if you like them.

I'll later add all the links to the documentation.

**Review by samjrholt — APPROVED** [review]

**Review by swapneelap — APPROVED** [review]
Looks good to me 👍

**Pull request merged**